warehouse/query-core/src/main/java/datawave/query/jexl/nodes/QueryPropertyMarker.java

```diff
@@ -161,6 +165,66 @@ protected void setupSource(JexlNode source) {
         this.jjtAddChild(refExpNode1, 0);
     }
     
+    /**
+     * Determines if this source node is already marked with the provided marker class
+     *
+     * @param source
+     *            a JexlNode
+     * @param markerClass
+     *            an instance of a {@link QueryPropertyMarker}
+     * @return true if this node is already marked with the provided marker class
+     */
+    public static boolean isSourceAlreadyMarked(JexlNode source, Class<? extends QueryPropertyMarker> markerClass) {
```

Comment:
I think this can be deleted in favor of just using `isSubTreeAlreadyMarked()`. See comment for that method.

Reply:
Resolving per our other conversation